fendermint/vm/interpreter/src/bytes.rs

```diff
@@ -31,6 +31,73 @@ impl<I> BytesMessageInterpreter<I> {
     }
 }
 
+#[async_trait]
+impl<I> ProposalInterpreter for BytesMessageInterpreter<I>
+where
+    I: ProposalInterpreter<Message = ChainMessage>,
+{
+    type State = I::State;
+    type Message = Vec<u8>;
+
+    /// Parse messages in the mempool and pass them into the inner `ChainMessage` interpreter.
+    async fn prepare(
+        &self,
+        state: Self::State,
+        msgs: Vec<Self::Message>,
+    ) -> anyhow::Result<Vec<Self::Message>> {
+        // We could include a flag to disable this parsing if we know that the inner interpreter
+        // does not care about proposals, but in our case we know that we'll eventually want to
+        // consult the IPLD Resolver.
+
+        let mut chain_msgs = Vec::new();
+
+        for msg in msgs {
+            match fvm_ipld_encoding::from_slice::<ChainMessage>(&msg) {
+                Err(e) => {
+                    // This should not happen because the `CheckInterpreter` implementation below would
+                    // have rejected any such user transaction.
+                    tracing::warn!(
+                        error = e.to_string(),
+                        "failed to decode message in mempool as ChainMessage"
+                    );
+                }
+                Ok(msg) => chain_msgs.push(msg),
+            }
+        }
+
+        let chain_msgs = self.inner.prepare(state, chain_msgs).await?;
+
+        chain_msgs
+            .into_iter()
+            .map(|msg| {
+                fvm_ipld_encoding::to_vec(&msg).context("failed to encode ChainMessage as IPLD")
+            })
+            .collect()
+    }
+
+    /// Parse messages in the block, reject if unknown format. Pass the rest to the inner `ChainMessage` interpreter.
+    async fn process(&self, state: Self::State, msgs: Vec<Self::Message>) -> anyhow::Result<bool> {
+        let mut chain_msgs = Vec::new();
+
+        for msg in msgs {
+            match fvm_ipld_encoding::from_slice::<ChainMessage>(&msg) {
+                Err(e) => {
+                    // This would indicate a Byzantine validator which includes rubbish in their proposal.
+                    // We could reject the proposal here, or we can accept it and punish the validator during
+                    // block execution, so that their power is reduced.
+                    tracing::debug!(
```

Comment:
should we reject the proposal here is we cannot deserialize the txn?

Reply:
Added the flag in https://github.com/consensus-shipyard/fendermint/pull/185/commits/a36603e957a58853dbe10af1053b309e8d92664a

Reply:
One more problem with rejection is that we have to be careful not to reject something we accidentally admitted into our own mempool, otherwise it will never be removed from it.